src/index.ts

```diff
@@ -632,11 +636,7 @@ export class KycDao extends ApiBase {
         this.session = undefined;
       }
     } else if (this.session) {
-      const isSameWallet =
-        this.session.chain_and_address.address === this._chainAndAddress?.address &&
-        this.session.chain_and_address.blockchain === this._chainAndAddress?.blockchain &&
-        this.session.chain_and_address.blockchainNetwork ===
-          this._chainAndAddress?.blockchainNetwork;
+      const isSameWallet = isEqual(this.session.chain_and_address, this._chainAndAddress);
```

Comment:
I know we discussed this briefly, but I really think this should be a simple helper method and not lodash's mega-universal comparision-cannon :/

Especially since we perfectly know the type of `chain_and_address`. Neither `MintingData` nor `ChainAndAddress` will magically grew a nested deep self-referential object graph. At least I hope so!

Sorry for not voicing this sooner.

I think adding a helper function to utils.ts is the way to go. (Which throws an error if it encounters a function or anything unexpected.)

My main problem with lodash's is that it's unmaintained, and it's not really universal either. ([deep-equal](https://github.com/inspect-js/node-deep-equal/blob/master/index.js) is, but it's 12K + deps + supports IE  ... wtf :D)

How about something like this?

```
const compare = (obj1: any, obj2: any): boolean => {
    if (typeof obj1 !== typeof obj2) return false;
    if (Object.keys(obj1).length !== Object.keys(obj2).length) return false;

    for (let key in obj1) {
        if (obj1.hasOwnProperty(key)) {
            if (!obj2.hasOwnProperty(key)) {
                return false;
            }

            if (typeof obj1[key] === 'object') {
                if (!compare(obj1[key], obj2[key])) return false;
            } else {
                if (obj1[key] !== obj2[key]) return false;
            }

        }
    }

    return true;
}

// chatgpt to the rescue
```

Reply:
Ok, I'll take a look.

Reply:
I replaced `lodash.isEqual` with a simpler implementation that should be fine for us for now. It can't compare everything that `deep-equal` can but I did some testing and it seems to be enough for nested objects and primitives and can be extended to be able to handle some other things.